### DISCUSSION TITLE
Can a failed DAG task force all running tasks to stop immediately?

### DISCUSSION THREAD

**Discussion by csm-kb** [argoproj/argo-workflows#13359] [Q&A]
Question above! Core context:
- My application contains two pods that talk to one another one-way through a messaging interface (think sensor information to a processing service).
- I have written an Argo workflow that performs an end-to-end test against specified versions of both of these pods. They spawn up, talk to each other, and then resolve successfully.
- However, if an error causes the sensor pod to fail abruptly, the processing service runs as normal, waiting for a termination message that will never arrive...

 ℹ I'm looking to stop the processing DAG task when the other branch fails, so the workflow can resolve `onExit` and fail gracefully instead of being left uncaught (except via e.g. long CI times).
- Is this currently possible with Argo?

From what I've read, I don't think so -- `FailFast` only skips scheduling of further DAG tasks, and a deep dive of issues and discussions didn't reveal anyone asking this exact "stop the whole DAG immediately if one task fails" question.

I am aware that I could incorporate some odd logic into my pods or write a sidecar to handle this testing edge-case -- but this _could_ be a beneficial Argo DAG config setting, if this isn't currently possible and there are other use cases that support it!

**Comment by Reversaidx** [comment]
have the same question
